### PR TITLE
llvmPackages_git.*: Bump to newer commit

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Not sure whether we want to merge this now, or wait for more patches to land upstream first, but it was good to at least confirm the CMake is still working

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sternenseemann — COMMENTED** [review]
The question whether to update or not to, I'd say depends on how much hassle it is for you to rebase the patches etc.?

Haven't looked at the patches or tried building yet, can take a closer look later.

File: pkgs/development/compilers/llvm/git/libcxx/default.nix [hunk 1]
```diff
@@ -10,12 +12,26 @@
...
+in
+
 stdenv.mkDerivation rec {
-  pname = if headersOnly then "cxx-headers" else "libcxx";
+  pname = basename + lib.optionalString headersOnly "-headers";
   inherit version;
 
-  inherit src;
-  sourceRoot = "source/libcxx";
+  src = runCommand "${pname}-src-${version}" {} ''
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/development/compilers/llvm/git/libcxx/default.nix:23]: Yeah I think that would be good, just didn't do it yet. I did something similar in https://github.com/NixOS/nixpkgs/pull/82131 i should perhaps pull out.

File: pkgs/development/compilers/llvm/git/llvm/default.nix [hunk 2]
```diff
@@ -22,7 +23,7 @@
   || stdenv.isAarch32 # broken for the armv7l builder
 )
 , enablePolly ? false
-}:
+} @args:
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/development/compilers/llvm/git/llvm/default.nix:26]: Ah thanks, yeah that was before I renamed `src` to `monorepoSrc`.

**Comment by primeos** [comment]
Thanks a lot @Ericson2314 and @sternenseemann! :) Unfortunately, `llvmPackages_git` was getting quite outdated.

> Not sure whether we want to merge this now, or wait for more patches to land upstream first, but it was good to at least confirm the CMake is still working

I wouldn't mind merging it. The patches situation isn't ideal (not to say can be a bit frustrating at times :P) but I guess for LLVM it should be fine (and this doesn't make it worse anyway, it just adds a bit more content to Nixpkgs). Hopefully the rest of the big patches will land soon... :)

Anyway: @Ericson2314 and @sternenseemann wouldn't you be interested to add yourself officially as LLVM package maintainers? AFAIK you're currently doing by far most of the work anyway and usually end up reviewing other PRs as well. So it seems like that would really make sense, unless you don't want to for some reason. (AFAIK most listed LLVM maintainers aren't that active btw, e.g., I'm only in there since I ended up having to add new versions for Chromium.)

**Comment by Ericson2314** [comment]
Yeah I think adding us would make sense.

File: pkgs/development/compilers/llvm/git/libcxx/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (sternenseemann) [reply pkgs/development/compilers/llvm/git/libcxx/default.nix:23]: Yeah, that looks right!

**Comment by primeos** [comment]
Can we merge this now so that we can package [LLVM 14.0.0-rc1](https://github.com/llvm/llvm-project/releases/tag/llvmorg-14.0.0-rc1)? (Probably makes sense to pick a revision that doesn't require additional changes for 14.0.0-rc1 (e.g., that exact revision or the commit where LLVM 14 was branched off from master) but maybe that's already the case for the current revision.)

I'm a bit late as it looks like it's already required for Chromium M99 which will be released on Mar 1.
Currently the build generates a binary that fails due to "Program received signal SIGILL, Illegal instruction." I haven't looked into that yet but a newer LLVM version could(/should) fix it.

**Comment by primeos** [comment]
`llvmPackages_git.stdenv` builds with https://github.com/NixOS/nixpkgs/pull/162101 but some other packages are broken - it'll have to do for now.

File: pkgs/development/compilers/llvm/git/llvm/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/development/compilers/llvm/git/llvm/default.nix:26]: Removing it in https://github.com/NixOS/nixpkgs/pull/268812

**Pull request merged**